cookbook/rag_examples/rag_prompts/news_rag_prompt.py

```diff
@@ -0,0 +1,98 @@
+import abc
+import inspect
+import os
+
+import pandas as pd
+from config import PINECONE_INDEX, TEXT_COLUMN
+from pinecone import Pinecone
+from pydantic import ConfigDict
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from utils import query_dataframe, query_pinecone
+
+from mirascope import OpenAIChat, Prompt, messages
+
+
+class Settings(BaseSettings):
+    model_config = SettingsConfigDict(env_file=".env")
+
+    OPENAI_API_KEY: str
+    PINECONE_API_KEY: str
+
+
+@messages
+class NewsRagPrompt(Prompt, abc.ABC):
+    """
+    SYSTEM:
+    You are an expert at:
+    1) determining the relevancy of articles to a topic, and
+    2) summarizing articles concisely and eloquently.
+
+    When given a topic and a list of possibly relevant texts, you determine for each
+    text if it is truly relevant to the topic, and only if so, do you summarize it. You
+    format your responses as only a list, where each item is a summary of an article or
+    an explanation as to why it is not relevant.
+
+    USER:
+    Here are {num_statements} article snippets about this topic: {topic}
+
+    {context}
+
+    Pick only the snippets which are truly relevant to the topic, and summarize them.
+    """
+
+    num_statements: int
+    topic: str
+    df: pd.DataFrame
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @property
+    @abc.abstractmethod
+    def context(self) -> str:
+        """Finds most similar articles to topic using embeddings."""
+        ...
+
+
+class LocalNewsRagPrompt(NewsRagPrompt):
```

Comment:
It's interesting, although it makes sense to group these prompts together given that they are very similar, it makes versioning one of the prompts much harder. I would recommend splitting these into three different prompts: `news_rag_base_prompt.py`, `local_news_rag_prompt.py`, and `pinecone_news_rag_prompt.py`

This solves a problem of clashing changes where you might make changes to local and pinecone but then want to revert just one of the changes and can't because you can't mix the different version easily.